### PR TITLE
Add wasm-bindgen

### PR DESCRIPTION
**Pull request by Ekleog**
Hey! I need to use wasm-bindgen-test-runner in CI, and thought I'd contribute back implementation of it.

Just to say, the `wasm-bindgen-cli` package actually includes other binaries (wasm-bindgen and wasm2es6js). I'm not a user of these packages and I don't know how useful they'd be in CI, so I didn't focus on them, but… it looks like in order to download multiple packages from a single archive, the user would have to download the archive one time for each binary, and we'd have to maintain the json file once for each binary?

So the main thing I'm unsure of is the naming of the tool. Should I have it be `wasm-bindgen-test-runner`, as there currently can be a single executable per tool? Or should we first support multiple executables per tool and then name it `wasm-bindgen-cli`, to match the way it can currently be installed via `binstall`? (I'll say that I probably won't have time to do the second option)

Also I'll take advantage to ask: I also wanted to integrate `sqlx-cli`, but AFAICT [its releases do not have binaries published](https://github.com/launchbadge/sqlx/releases/tag/v0.7.3). Do I understand correctly that install-action is thus unable to help here?

### CODE REVIEW

**Comment by Ekleog** [comment]
I think this is ready for review! :D There is just the `tidy` test that is failing, but it is not related to this PR and seems to be due to the way CI works. I tried simply moving `main.sh` to `tools/`, but then other things started failing. Not knowing whether that is actually the way you want to go, I just gave up on fixing CI and left this one red :)

File: main.sh
Comment (NobodyXu) [inline main.sh:670]: I think the space added before comment could be the reason why tidy has failed

File: tools/codegen/Cargo.toml
Comment (NobodyXu) [inline tools/codegen/Cargo.toml:3]: I don't think this internal crate should have a version, rather it should have:

```suggestion
publish = false
```

**Review by NobodyXu — CHANGES_REQUESTED** [review]

File: tools/codegen/Cargo.toml
Comment (taiki-e) [reply tools/codegen/Cargo.toml:3]: Neither field is needed for version-less manifests (https://github.com/rust-lang/cargo/pull/12786).

File: main.sh
Comment (Ekleog) [reply main.sh:670]: Actually I'm pretty sure tidy has failed due to this line:
```
Error: file-permissions-check failed: executable should be in tools/ directory
```

I added the spaces when tidy told me to :)

File: tools/codegen/Cargo.toml
Comment (Ekleog) [reply tools/codegen/Cargo.toml:3]: I can remove it if you want, but my local distribution-provided version of cargo failed due to lack of the `version` field.

TBH I don't really care, I just added it en-passant, thinking it could help other people have one fewer thing to think about when contributing :)

File: main.sh
Comment (taiki-e) [reply main.sh:670]: > ```
> Error: file-permissions-check failed: executable should be in tools/ directory
> ```

This is due to the fact that you have added a version to codegen. See https://github.com/taiki-e/install-action/issues/302.

File: tools/codegen/Cargo.toml
Comment (taiki-e) [reply tools/codegen/Cargo.toml:3]: To be honest, I'm surprised that codegen worked with the old rustc.

We are basically not going to guarantee that codegen will work on older rustc. (If you want to do that you need to check it with CI, otherwise it makes no sense as it can be easily broken.)

**Comment by Ekleog** [comment]
Ooh I see, thank you for your answers!

I just removed the drive-by addition of the `version` field, if as you say codegen isn't supposed to work with older rustc versions and version-less manifests are not going away anyway :)

Now CI passed, please let me know if you think there are other changes to make to this PR!

The main question I have is still how to deal with the multiple-binaries-inside-a-single-crate situation: currently it's possible to install things via `wasm-bindgen-cli` which will proxy through `binstall`, but AFAICT I can't just add a manifest that'd install multiple binaries, so I can't just name the tool `wasm-bindgen-cli` :/

**Review by NobodyXu — APPROVED** [review]
LGTM, but I'd let taiki-e to merge this

**Comment by taiki-e** [comment]
LGTM assuming we merge this with a single binary. However, I would like to do a little evaluation to see how difficult it would be to support multiple binaries before merging. (We have already done something more complicated for protoc but that requires some manually written code.)

**Comment by taiki-e** [comment]
Ok, I pushed a commit to support multiple binaries:

```
info: installing wasm-bindgen@latest
info: downloading https://github.com/rustwasm/wasm-bindgen/releases/download/0.2.91/wasm-bindgen-0.2.91-x86_64-unknown-linux-musl.tar.gz
info: verifying sha256 checksum for wasm-bindgen-0.2.91-x86_64-unknown-linux-musl.tar.gz
info: wasm-bindgen-test-runner installed at /home/runner/.cargo/bin/wasm-bindgen-test-runner
info: wasm-bindgen installed at /home/runner/.cargo/bin/wasm-bindgen
info: wasm2es6js installed at /home/runner/.cargo/bin/wasm2es6js
+ wasm-bindgen --version
wasm-bindgen 0.2.91 (fe8bc949b)
```

**Comment by taiki-e** [comment]
Published in 2.28.0. Thanks for the PR!

**Comment by Ekleog** [comment]
Awesome, thank you for all the improvements you did to support this use case!

**Pull request merged**
